Option 1 - Create dictionary program

vishaltiparadi opened the pull request:
Option 1 - Create Dictionary program

vishaltiparadi committed:
Create dictionary.cpp

ashutosh-bapat commented:
There are couple of problems here
1. The implementation uses an abstract data structure Set without any implementation details. It doesn't list any assumptions about the implementation of Set as well. It will be good to assume some non-abstract data structures like HashTable/LinkedList and add assumptions about their implementation.
2. The program doesn't specify when and how the shared memory is created and when it gets destroyed.

vishaltiparadi committed:
Update dictionary.cpp

The first version of code has been overwritten in order to address the 2 comments.
I understand this code can still be refined further. Also, this is a workable code, which can be compiled and tested with different options.
Usage-
1. to insert - exename insert word
2. to search - exename search word
3. to delete - exename delete word
4. to destroy shared memory - exename exit anyword.
I know that exit command doesn't need any word as such, but this needs to be revised.
Thanks.
